src/client/auth/oidc.rs

```diff
@@ -279,16 +408,45 @@ pub(crate) async fn reauthenticate_stream(
 ) -> Result<()> {
     credential
         .oidc_callback
-        .as_ref()
-        .unwrap()
-        .cache
+        .inner
         .lock()
         .await
+        .as_mut()
+        .unwrap()
+        .cache
         .invalidate(conn, true)
         .await;
     authenticate_stream(conn, credential, server_api, None).await
 }
 
+#[cfg(feature = "azure-oidc")]
+async fn setup_automatic_providers(
+    credential: &Credential,
+    guard: &mut MutexGuard<'_, Option<StateInner>>,
```

Comment:
Does this need to take a `MutexGuard` or can it take an `Option<&mut StateInner>`?

Reply:
I think it should work either way. I've never cared too much with non user facing interfaces, is all.

Reply:
hm, there doesn't seem to be away to assign to it that way though.

Reply:
`&mut Option<StateInner>` works, however.